ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java

```diff
@@ -58,6 +59,15 @@ private void tryEvaluatingPendingInput(TableRuntime tableRuntime, ArcticTable ta
     }
   }
 
+  @Override
+  public void handleConfigChanged(TableRuntime tableRuntime, TableConfiguration originalConfig) {
+    // After disabling self-optimizing, use dispose() to change the status of the table to IDLE
+    if (originalConfig.getOptimizingConfig().isEnabled()
+        && !tableRuntime.getTableConfiguration().getOptimizingConfig().isEnabled()) {
+      tableRuntime.dispose();
```

Comment:
It's not quite appropriate to call this `dispose` method here, which is designed to clear the table runtime after table is deleted.

I think after detecting optimizing not enabled, we just need to cancel the current running optimizing-process(`com.netease.arctic.server.table.TableRuntime#optimizingProcess`) and set it to null.

Reply:
Yes, we shouldn't clear the table runtime after table is deleted.

fixed.